neuralmonkey/decoders/multi_decoder.py

```diff
@@ -0,0 +1,134 @@
+import math
+import tensorflow as tf
+import numpy as np
+
+from neuralmonkey.vocabulary import START_TOKEN
+from neuralmonkey.decoding_function import attention_decoder
+from neuralmonkey.logging import log
+from neuralmonkey.dataset import Dataset
+
+
+class MultiDecoder(object):
+    """A class that manages parts of the computation graph that are
```

Comment:
Here, I would expect a thorough explanation what this class does and why would anyone need that.